pil-analyzer/src/pil_analyzer.rs

```diff
@@ -326,15 +323,52 @@ impl PILAnalyzer {
         }
     }
 
-    fn resolve_trait_impls(&mut self) {
-        self.solved_impls =
-            traits_resolution(&self.definitions, &self.identities, &self.implementations);
+    fn resolve_trait_impls(&mut self) -> HashMap<String, HashMap<Vec<Type>, Arc<Expression>>> {
+        let mut references = Vec::new();
+
+        let resolve_references = |expr: &Expression| {
+            expr.all_children()
+                .filter_map(|expr| {
+                    if let Expression::Reference(
+                        _,
+                        Reference::Poly(
+                            reference @ PolynomialReference {
+                                type_args: Some(_), ..
+                            },
+                        ),
+                    ) = expr
+                    {
+                        Some(reference.clone())
```

Comment:
Why do you need to clone here? Would it work for the `resolve_references` function to return an iterator (and not converting the iterator to a vector at all)?